numpy/core/src/multiarray/ctors.c

```diff
@@ -1876,7 +1882,17 @@ PyArray_GetArrayParamsFromObject(PyObject *op,
             return 0;
         }
         /* If object arrays are forced */
-        if (is_object) {
+        if (is_object != DISCOVERED_OK) {
+            if (is_object == DISCOVERED_RAGGED && requested_dtype == NULL) {
+                /* NumPy 1.18, 2019-11-01 */
```

Comment:
If this isn't making the release...
```suggestion
                /* NumPy 1.19, 2019-11-01 */
```

Reply:
It is waiting on your review. It can be in the release if you approve it soon, but is not a blocker.

Reply:
Added a comment above - a single test in `test_deprecation.py` is enough to make me happy, no need to rewrite all the other tests.